src/main/java/igrad/model/CourseBook.java

```diff
@@ -78,16 +82,20 @@ public void resetData(ReadOnlyCourseBook newData) {
     /**
      * Adds the given courseInfo (only one courseInfo can exist/ever be created in the system).
      */
-    public void addCourseInfo(CourseInfo c) {
-        // TODO: Restrict to one
+    public void addCourseInfo(CourseInfo c) throws CommandException {
+        if (firstSet == false) {
```

Comment:
Hi I think this one instead of using the boolean, maybe you can just check if courseinfo already has a non-empy name! :)

Reply:
hmm yeah, i think i'll use Optional

Reply:
in fact, `Name` is already optional haha

Reply:
thanks yijie, for pointing  out